src/main/java/com/mongodb/spark/sql/connector/connection/MongoConnectionProvider.java

```diff
@@ -82,5 +144,56 @@ public void doWithCollection(
         mongoDatabase ->
             function.apply(mongoDatabase.getCollection(collectionName, BsonDocument.class)));
   }
-  // CHECKSTYLE.ON:JavadocMethod
+
+  /** @return memoize's and returns the {@link MongoClientCache} */
+  private static synchronized MongoClientCache getMongoClientCache() {
```

Comment:
Creates the singleton cache if it doesn't exist.

Uses the same system property as the existing connector.